decode-audio-promise/index.html

```diff
@@ -0,0 +1,130 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width" />
+
+    <title>Web Audio API examples: decodeAudioData()</title>
+  </head>
+
+  <body>
+    <h1>Web Audio API examples: decodeAudioData()</h1>
+
+    <button class="play">Play</button>
+    <button class="stop">Stop</button>
+
+    <h2>Set playback rate</h2>
+    <input
+      class="playback-rate-control"
+      type="range"
+      min="0.25"
+      max="3"
+      step="0.05"
+      value="1"
+    />
+    <span class="playback-rate-value">1.0</span>
+
+    <h2>Set loop start and loop end</h2>
+    <input
+      class="loopstart-control"
+      type="range"
+      min="0"
+      max="20"
+      step="1"
+      value="0"
+    />
+    <span class="loopstart-value">0</span>
+
+    <input
+      class="loopend-control"
+      type="range"
+      min="0"
+      max="20"
+      step="1"
+      value="0"
+    />
+    <span class="loopend-value">0</span>
+  </body>
+  <script>
+    window.addEventListener("DOMContentLoaded", loadPage, false);
+
+    // define variables
+    const audioCtx = new AudioContext();
+    let buffer;
+    let source;
+
+    const play = document.querySelector(".play");
+    const stop = document.querySelector(".stop");
+
+    const playbackControl = document.querySelector(".playback-rate-control");
+    const playbackValue = document.querySelector(".playback-rate-value");
+    playbackControl.setAttribute("disabled", "disabled");
+
+    const loopstartControl = document.querySelector(".loopstart-control");
+    const loopstartValue = document.querySelector(".loopstart-value");
+    loopstartControl.setAttribute("disabled", "disabled");
+
+    const loopendControl = document.querySelector(".loopend-control");
+    const loopendValue = document.querySelector(".loopend-value");
+    loopendControl.setAttribute("disabled", "disabled");
+
+    function loadPage() {
+      fetchAudio("viper").then((buf) => { // executes when buffer has been decoded
+          let max = Math.floor(buf.duration); // in this case buf === global buffer
+          loopstartControl.setAttribute("max", max);
+          loopendControl.setAttribute("max", max);
+      });
+    }
+
+    // use fetch() to load an audio track, and
+    // decodeAudioData to decode it and stick it in global buffer variable.
+    // We put the buffer into the source in play.onclick()
+    async function fetchAudio(name) {
+      try {
+        let rsvp = await fetch(`${name}.mp3`);
+        buffer   = await audioCtx.decodeAudioData(await rsvp.arrayBuffer());
+        return buffer; // returns a Promise, buffer is arg for .then((arg) => {})
+      }
+      catch(err) {
+        console.log(`Unable to fetch the audio file: ${name}\n\
+                     error: ${err.message}`);
```

Comment:
```suggestion
        console.log(`Unable to fetch the audio file: ${name}
error: ${err.message}`);
```

Reply:
Thanks for relieving me of a misapprehension about newlines and whitespace.  I was conflating it with html.